dhis-2/dhis-support/dhis-support-db-migration/src/main/java/org/hisp/dhis/db/migration/v34/V2_34_22__Add_teav_btree_index.java

```diff
@@ -0,0 +1,28 @@
+package org.hisp.dhis.db.migration.v34;
+
+import org.flywaydb.core.api.migration.BaseJavaMigration;
+import org.flywaydb.core.api.migration.Context;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.sql.SQLException;
+import java.sql.Statement;
+
+public class V2_34_22__Add_teav_btree_index extends BaseJavaMigration
+{
+    private static final Logger log = LoggerFactory.getLogger( V2_34_22__Add_teav_btree_index.class );
+
+    @Override
+    public void migrate( Context context )
+        throws Exception
+    {
+        try ( Statement statement = context.getConnection().createStatement() )
+        {
+            statement.execute( "create index in_attribute_value on trackedentityattributevalue using btree (trackedentityattributeid, lower(value)) " );
```

Comment:
I think we need to enforce a max length (e.g. 1200) on TEI attribute values in the same PR to avoid too long values failing to be persisted.